enhancements/update/upgrade-without-registry.md

```diff
@@ -0,0 +1,989 @@
+the technicians, together with the documentation explaining how to use it.
+
+#### Documentation to use the bundle
+
+This documentation shouldn't assume previous OpenShift knowledge, and should be
+basic instructions to plug the USB stick containing the bundle and then check if
+upgrade the upgrade succeeded or failed.
+
+### Risks and Mitigations
+
+The proposed solution will require space to store the release bundle and all the
+release images in all the nodes of the cluster, approximately 48 GiB in the
+worst case. To mitigate that risks the components that will consume disk space
+will check in advance if the required space is available.
+
+### Drawbacks
+
+This approach requires non trivial changes to the cluster version operator and,
+to a lesser degree, to the machine config operator.
+
+## Design Details
+
+### Open Questions
+
+None.
+
+### Test Plan
+
+We should have at least tests that verifies that the upgrade can be performed in
+a fully disconnected environment, both for a single node cluster and a cluster
+with multiple nodes. These tests should gate the OCP release.
+
+It is desirable to have another test that scans the OCP components looking for
+use of the `Always` pull policy. This should probably run for each pull request
+of each OCP component, and prevent merging if it detects that the offending pull
+policy is used. We should consider adding admission in CI for this.
+
+### Graduation Criteria
+
+The feature will ideally be introduced as `Dev Preview` in OpenShift 4.14.z,
+moved to `Tech Preview` in 4.15 and declared `GA` in 4.16.
+
+#### Dev Preview -> Tech Preview
+
+- Ability to upgrade a single-node clusters using the `bundle.enable: false` mode
+(no bundle, just pinning and pre-loading of the images).
+
+- Availability of the tests that verify the upgrade if single-node clusters.
+
+- Availability of the tests that verify that no OCP component uses the `Always`
+pull policy.
+
+- Obtain positive feedback from at least one customer.
+
+#### Tech Preview -> GA
+
+- Ability to upgrade single-node and multi-node clusters using the
+`bundle.enable` set to `true` or `false`: with a bundle or just pinning and
+pre-loading the images.
+
+- Ability to create bundles using the `oc adm upgrade create bundle` command.
+
+- Availability of the tests that verify the upgrade in single-node and
+multi-node clusters.
+
+- User facing documentation created in
+[https://github.com/openshift/openshift-docs](openshift-docs).
+
+#### Removing a deprecated feature
+
+Not applicable, no feature will be removed.
+
+### Upgrade / Downgrade Strategy
+
+There are no additional considerations for upgrade or downgrade. The same
+considerations that apply to the cluster version operator in general will also
+apply in this case.
+
+### Version Skew Strategy
+
+This feature will only be usable once the cluster version operator and the
+machine config operator have been upgraded to support it. That upgrade will have
+to be done by other means.
```

Comment:
So if I'm in a disconnected environment right now, how do I upgrade? Why is this not sufficient for the long term?

Reply:
My understanding is that currently the only supported way to upgrade in a disconnected environment is to use a registry mirror. The upgrade that brings the new capability will have to be done using that registry mirror, or else re-installing the cluster.

Reply:
Can you make sure this is mentioned somewhere, it would be good to understand why this new solution is better than the existing/old solution but I don't think it's mentioned anywhere yet so is hard to compare

Reply:
This solution is not in general better than the current one. It is better specifically for disconnected clusters where it isn't possible or desirable to bring up an external mirror registry. The "Alternatives" section explains that. Does it seem enough or should we expand it?

Reply:
I think it would be good to update the alternatives to explain that the alternatives mentioned there are the currently supported solutions, so, the alternative effectively is do nothing and maintain the status quo. It wasn't obvious that those alternatives were the current solution

Reply:
I tried to clarify that in the alternatives section.